### PR TITLE
yuzu: Implement basic controller UI navigation

### PR DESCRIPTION
**Pull request by german77**
This is one of the most requested features. Adds the ability to navigate through the game list, profile selector and controller applet.

The controller applet still needs some work since it's a bit more complex.

### CODE REVIEW

File: src/yuzu/controller_navigation.cpp [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                                         Settings::NativeButton::Values native_button,
+                                         Qt::Key key) {
+    if (buttons[native_button].value && !button_values[native_button].locked) {
+        emit TriggerKeyboardEvent(key);
+    }
+}
+
+void ControllerNavigation::ControllerUpdateEvent(Core::HID::ControllerTriggerType type) {
+    if (type == Core::HID::ControllerTriggerType::Button) {
+        const auto controller_type = controller->GetNpadStyleIndex();
```
Comment (lioncash) [inline src/yuzu/controller_navigation.cpp:41]: Maybe move the contents of this if into a `ControllerUpdateButton()` member function and call it here, and do the same for the sticks as well. That way their behavior is kept separate from one another

File: src/yuzu/controller_navigation.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+private:
+    void TriggerButton(const Core::HID::ButtonValues& buttons,
+                       Settings::NativeButton::Values native_button, Qt::Key key);
+    void ControllerUpdateEvent(Core::HID::ControllerTriggerType type);
+
+    Core::HID::ButtonValues button_values{};
+    Core::HID::SticksValues stick_values{};
+
+    int callback_key;
```
Comment (lioncash) [inline src/yuzu/controller_navigation.h:39]:
```suggestion
    int callback_key{};
```

even if it's overwritten in the constructor, so it has a consistent value, regardless of which phase the constructor is in for debugging purposes.

File: src/yuzu/controller_navigation.h [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+// Refer to the license.txt file included
+
+#pragma once
+
+#include <QKeyEvent>
+#include <QObject>
+
+#include "common/input.h"
+#include "common/settings.h"
+#include "core/hid/emulated_controller.h"
```
Comment (Morph1984) [inline src/yuzu/controller_navigation.h:12]: This can be a forward declaration

File: src/yuzu/game_list.h [hunk 4]
```diff
@@ -23,6 +23,7 @@
 #include "common/common_types.h"
 #include "uisettings.h"
 #include "yuzu/compatibility_list.h"
+#include "yuzu/controller_navigation.h"
```
Comment (Morph1984) [inline src/yuzu/game_list.h:26]: This can be a forward declaration

**Comment by Morph1984** [comment]
I would suggest moving controller_navigation to the `util` folder

File: src/yuzu/util/controller_navigation.h [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+// Refer to the license.txt file included
+
+#pragma once
+
+#include <QKeyEvent>
+#include <QObject>
+
+#include "common/input.h"
+#include "common/settings.h"
+#include "core/hid/emulated_controller.h"
```
Comment (Morph1984) [inline src/yuzu/util/controller_navigation.h:12]: This can be a forward declaration

**Review by Morph1984 — APPROVED** [review]

**Pull request merged**